mpqp/core/circuit.py

```diff
@@ -210,15 +212,23 @@ def add(self, components: OneOrMany[Instruction | NoiseModel]):
                 self.add(comp)
             return
 
-        if any(conn >= self.nb_qubits for conn in components.connections()):
-            component_type = (
-                "Instruction" if isinstance(components, Instruction) else "Noise model"
-            )
-            raise NumberQubitsError(
-                f"{component_type} {type(components)}'s connections "
-                f"({components.connections()}) are not compatible with circuit"
-                f" size ({self.nb_qubits})."
-            )
+        if self._user_nb_qubits is not None:
+            if any(conn >= self.nb_qubits for conn in components.connections()):
```

Comment:
Since `self._user_nb_qubits is not None`, you can directly use `self._user_nb_qubits` instead of calling the property `self.nb_qubits`